Bit-Manipulation/GenerateSubSets.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * @function generateSubSets
+ * @param {Array} inputArray
+ * @returns {Array}
+ * @example [1,2] -> [[],[1],[2],[1,2]]
+ */
+
+// The time complexity of this algorithm is BigO(2^n) where n is the length of array
+function generateSubSets(inputArray) {
+  if (!Array.isArray(inputArray)) {
+    return 'Input is not an array'
+  }
+  let arrayLength = inputArray.length
+  let subSets = []
```

Comment:
Could use `Array(1 << arrayLength)` to allocate this to the right size from the get go, then you could directly use `subSets[i] = subSet` below.